rest_lights.cpp

```diff
@@ -2975,30 +2975,32 @@ void DeRestPluginPrivate::handleLightEvent(const Event &e)
                     {
                         iy = item;
                     }
-                    else if (item->lastSet().isValid() && (gwWebSocketNotifyAll || (item->lastChanged().isValid() && item->lastChanged() >= lightNode->lastStatePush)))
+                    else if (item->lastSet().isValid() && (gwWebSocketNotifyAll || (item->lastChanged().isValid() && item->needPushChange())))
                     {
                         state[key] = item->toVariant();
+                        item->clearNeedPush();
                     }
                 }
             }
 
             if (ix && ix->lastSet().isValid() && iy && iy->lastSet().isValid())
             {
                 if (gwWebSocketNotifyAll ||
-                    (ix->lastChanged().isValid() && ix->lastChanged() >= lightNode->lastStatePush) ||
-                    (iy->lastChanged().isValid() && iy->lastChanged() >= lightNode->lastStatePush))
+                    (ix->lastChanged().isValid() && ix->needPushChange()) ||
+                    (iy->lastChanged().isValid() && iy->needPushChange()))

```

Comment:
I think we don't need the `.lastChanged().isValid()` anymore. To my understanding when `.needPushChange()` is set, it implies that the timestamp is valid.

Reply:
OK, happy to move it, but this is older code, that this PR hasn't touched.

Reply:
Removed calls to `item->lastSet().isValid()` and `item->lastChanged().isValid()`.